frontend/src/Metamaps/JIT.js

```diff
@@ -765,71 +769,109 @@ const JIT = {
   onDragMoveTopicHandler: function (node, eventInfo, e) {
     var self = JIT
 
-    // this is used to send nodes that are moving to 
-    // other realtime collaborators on the same map
-    var positionsToSend = {}
-    var topic
-
     var authorized = Active.Map && Active.Map.authorizeToEdit(Active.Mapper)
 
     if (node && !node.nodeFrom) {
-      var pos = eventInfo.getPos()
+      var pos = eventInfo.getPos(),
+          EDGE_THICKNESS = 50 * Metamaps.Visualize.mGraph.canvas.scaleOffsetX,
+          SHIFT = 10 / Metamaps.Visualize.mGraph.canvas.scaleOffsetX,
+          PERIOD = 50;
+          
+      //self.virtualPointer = pos;
       // if it's a left click, or a touch, move the node
-      if (e.touches || (e.button == 0 && !e.altKey && !e.ctrlKey && !e.shiftKey && (e.buttons == 0 || e.buttons == 1 || e.buttons == undefined))) {
+      if (e.touches || (e.button == 0 && !e.altKey && !e.ctrlKey && (e.buttons == 0 || e.buttons == 1 || e.buttons == undefined))) {
+
+        var width = Metamaps.Visualize.mGraph.canvas.getSize().width,
+            height = Metamaps.Visualize.mGraph.canvas.getSize().height,
+            xPix = Metamaps.Util.coordsToPixels(pos).x,
+            yPix = Metamaps.Util.coordsToPixels(pos).y;
+            
+        if(self.dragFlag == 0){
+          self.mouseDownPix = Metamaps.Util.coordsToPixels(eventInfo.getPos());
+          self.dragFlag = 1;
+        }
+        
+        if(Metamaps.Util.getDistance(Metamaps.Util.coordsToPixels(pos),self.mouseDownPix) > 2 && !self.dragTolerance){
+          self.dragTolerance = 1;
+        }
+
+        if(xPix < EDGE_THICKNESS && !self.hasOwnProperty('dragLeftEdge') && self.dragTolerance ){
+          Metamaps.Visualize.mGraph.canvas.translate(SHIFT,0);
+          self.virtualPointer = {x:pos.x - SHIFT,y:pos.y};
+          self.updateTopicPositions(node, self.virtualPointer);
+          Visualize.mGraph.plot();
+          
+          self.dragLeftEdge = setInterval( function(){
+            Metamaps.Visualize.mGraph.canvas.translate(SHIFT,0);
+            self.virtualPointer = {x:self.virtualPointer.x - SHIFT,y:self.virtualPointer.y};
+            self.updateTopicPositions(node,self.virtualPointer);
+            Visualize.mGraph.plot();
+          } , PERIOD);
+        }
+        if(width - xPix < EDGE_THICKNESS && !self.hasOwnProperty('dragRightEdge') && self.dragTolerance){
+          Metamaps.Visualize.mGraph.canvas.translate(-SHIFT,0);
+          self.virtualPointer = {x:pos.x + SHIFT,y:pos.y};
+          self.updateTopicPositions(node, self.virtualPointer);
```

Comment:
nice work @Bortseb!
I tried out the implementation on your C9 instance and it works great, for the most part! I notice it can be a bit 'glitchy' visually, and I think there might be some kind of change that we can do to the above three lines to resolve the glitchy-ness. 

right now the approach is: 
translate the canvas
move the topic by the exact same amount as the canvas was translated. this is good in theory, but as we can see, glitchy in practice, possibly because of the setTimeouts. 

our other option is: 
translate the canvas
use `util.pixelsToCoords` to calculate what the position of the cursor over the canvas is equal to presently
use updateTopicPositions to update the node to the coordinate returned by the last function call

let's give that a try? 
I also think we can increase the dimension of the dragzones at the edges of the screen a little bit :)